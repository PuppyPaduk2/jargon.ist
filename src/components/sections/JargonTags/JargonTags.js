// @flow

import React, { Fragment } from 'react';
import Link from 'gatsby-link';
import cx from 'classnames';

import { type Tag } from '../../../types/tag';

import './JargonTags.scss';

type Props = {
  tags: Array<Tag>,
  className?: string,
  max?: number,
};

const JargonTags = ({ tags, className, max = 6 }: Props) => (
  <div className={cx('c-jargon-tags', className)}>
    <Fragment>
      {tags.slice(0, max).map(tag => (
        <Link key={tag.slug} to={`/k/${tag.slug}`} className="c-jargon-tags__tag u-gap-right-small">
          {tag.title}
        </Link>
      ))}
      {tags.length > max && (
        <span className="c-jargon-tags__more">ve {tags.length - max} tane daha</span>
      )}
    </Fragment>
  </div>
);

JargonTags.defaultProps = {
  className: '',
  max: 6,
};

export default JargonTags;
