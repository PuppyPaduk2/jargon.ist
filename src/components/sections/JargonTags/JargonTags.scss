@import '../../../styles/abstracts/index';

.c-jargon-tags {
  &__tag {
    background-color: $color-comet;
    border-radius: px-to-rem(2px);
    box-shadow: 0 px-to-rem(2px) px-to-rem(4px) 0 rgba(0, 0, 0, 0.1);
    color: $color-white;
    font-size: $g-font-size-small;
    padding: pad(2xsmall);

    &:hover {
      background-color: darken($color-comet, 5);
      color: $color-white;
      text-decoration: none;
    }
  }

  &__more {
    color: $color-comet;
    font-size: $g-font-size-small;
    padding: pad(2xsmall);
  }
}
