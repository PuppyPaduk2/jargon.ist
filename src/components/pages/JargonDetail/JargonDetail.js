// @flow
/* eslint-disable react/no-danger */

import React from 'react';

import { Container, Section } from '../../ui';
import JargonTags from '../../sections/JargonTags';

import { type Jargon } from '../../../types/jargon';

type Props = {
  jargon: Jargon,
};

const JargonDetail = ({ jargon }: Props) => (
  <Section>
    <Container>
      <div className="u-pad-sides-2xlarge@lg-up u-pad-sides-large@md-up u-pad-sides@sm-up">
        <h1 className="u-color-primary u-clear-gap">{jargon.title}</h1>
        {(jargon.tags || []).length > 0 && (
          <JargonTags className="u-gap-top-small" tags={jargon.tags || []} />
        )}
        <div className="u-pad-top" dangerouslySetInnerHTML={{ __html: jargon.html }} />
      </div>
    </Container>
  </Section>
);

export default JargonDetail;

/* eslint-enable react/no-danger */
