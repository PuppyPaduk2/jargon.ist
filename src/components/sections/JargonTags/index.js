// @flow

import JargonTags from './JargonTags';

export default JargonTags;
