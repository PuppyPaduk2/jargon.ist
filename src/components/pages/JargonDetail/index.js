// @flow

import JargonDetail from './JargonDetail';

export default JargonDetail;
