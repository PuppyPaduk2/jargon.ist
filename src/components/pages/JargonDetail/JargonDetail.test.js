// @flow

import React from 'react';
import { shallow } from 'enzyme';

import jargonsFixture from '../../../../__fixtures__/jargons';

import JargonTags from '../../sections/JargonTags';

import JargonDetail from './JargonDetail';

it('renders correctly', () => {
  const wrapper = shallow(<JargonDetail jargon={jargonsFixture[0]} />);

  expect(wrapper).toMatchSnapshot();
});

it('renders link to tags', () => {
  const wrapper = shallow(<JargonDetail jargon={jargonsFixture[0]} />);

  expect(wrapper.find(JargonTags)).toHaveLength(1);
});
