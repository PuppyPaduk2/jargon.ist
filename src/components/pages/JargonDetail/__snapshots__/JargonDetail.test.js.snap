// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Section
  className=""
>
  <Container
    className=""
  >
    <div
      className="u-pad-sides-2xlarge@lg-up u-pad-sides-large@md-up u-pad-sides@sm-up"
    >
      <h1
        className="u-color-primary u-clear-gap"
      >
        Example Jargon 1
      </h1>
      <JargonTags
        className="u-gap-top-small"
        max={6}
        tags={
          Array [
            Object {
              "slug": "tag1",
              "title": "tag1",
            },
            Object {
              "slug": "tag2",
              "title": "tag2",
            },
          ]
        }
      />
      <div
        className="u-pad-top"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Hello world</p>",
          }
        }
      />
    </div>
  </Container>
</Section>
`;
